otelcollector/telegraf/telegraf-prometheus-collector-ds.conf

```diff
@@ -0,0 +1,229 @@
+###############################################################################
+[[processors.converter]]
+  [processors.converter.fields]
+    float = ["*"]
+
+# Transforms tag and field values as well as measurement, tag and field names with regex pattern
+[[processors.regex]]
+  namepass = ["opentelemetry_allocator_targets"]
+
+  # Tag and field conversions defined in a separate sub-tables
+  [[processors.regex.tags]]
+    ## Tag to change, "*" will change every tag
+    key = "job_name"
+    ## Regular expression to match on a tag value
+    pattern = '(^(podmonitor|servicemonitor)\/.*)|(.*)'
+    ## Matches of the pattern will be replaced with this string.  Use ${1}
+    ## notation to use the text of the first submatch.
+    replacement = "${2}"
+
+###############################################################################
+#                            AGGREGATOR PLUGINS                               #
+###############################################################################
+[[aggregators.quantile]]
+  period = "5m"
+  drop_original = true
+  quantiles = [0.50,0.95]
+  algorithm = "t-digest"
+  compression = 100.0
+  namepass = ["otelcollector", "metricsextension"]
+
+# Keep the aggregate basicstats of each metric passing through.
+[[aggregators.basicstats]]
+  namepass = ["opentelemetry_allocator_targets"]
+
+  ## The period on which to flush & clear the aggregator.
+  period = "5m"
+
+  ## If true, the original metric will be dropped by the
+  ## aggregator and will not get sent to the output plugins.
+  drop_original = true
+
+  ## Configures which basic stats to push as fields
+  stats = ["count"]
+ 
+###############################################################################
+#                            INPUT PLUGINS                                    #
+###############################################################################
+
+# Read metrics about cpu usage
+#[[inputs.cpu]]
+  ## Whether to report per-cpu stats or not
+#  percpu = false
+  ## Whether to report total system cpu stats or not
+#  totalcpu = true
+  ## If true, collect raw CPU time metrics.
+#  collect_cpu_time = false
+  ## If true, compute and report the sum of all non-idle CPU states.
+#  report_active = true
+#  fieldpass = ["usage_active","cluster","node","host","device"]
+#  taginclude = ["cluster","cpu","node"]
+  
+# Read metrics about memory usage
+#[[inputs.mem]]
+#  fieldpass = ["used_percent", "cluster", "node","host","device"]
+#  taginclude = ["cluster","node"]
+
+[[inputs.procstat]]
+   exe = "otelcollector"
+   interval = "10s"
+   pid_finder = "pgrep"
+   pid_tag = true
+   name_override = "otelcollector"
+   fieldpass = ["cpu_usage", "memory_rss"]
+   [inputs.procstat.tags]
+#    Computer = "$NODE_NAME"
+#    NodeIp = "$NODE_IP"
+    cpulimit = "$CONTAINER_CPU_LIMIT"
+    memlimit = "$CONTAINER_MEMORY_LIMIT"
+    
+    defaultscrapekubelet = "$AZMON_PROMETHEUS_KUBELET_SCRAPING_ENABLED"
```

Comment:
Are all these setting relevant to ds?

Reply:
No good point, we can also remove these, we only need them once from rs